packages/components/src/client/components/ArtPlayer.ts

```diff
@@ -0,0 +1,211 @@
+     */
+    poster: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.title
+     */
+    title: {
+      type: String,
+      default: "",
+      required: false,
+    },
+
+    /**
+     * artplayer src.muted
+     */
+    muted: {
+      type: Boolean,
+      default: false,
+      required: false,
+    },
+
+    /**
+     *  return artplayer
+     */
+    player: {
+      type: Function,
+      default: undefined,
+      required: false,
+    },
+
+    /**
+     * artplayer danmuku
+     */
+    pluginDanmuKu: {
+      type: Object,
+      default: () => ART_PLAYER.pluginDanmuKu || {},
+      required: false,
+    },
+
+    /**
+     * Component width
+     *
+     * 组件宽度
+     */
+    width: {
+      type: [String, Number],
+      default: ART_PLAYER?.width || optionsDefault.width || undefined,
+      required: false,
+    },
+
+    /**
+     * Component height
+     *
+     * 组件高度
+     */
+    height: {
+      type: [String, Number],
+      default: ART_PLAYER?.height || undefined,
+      required: false,
+    },
+
+    /**
+     * Component width / height ratio
+     *
+     * 组件长宽比
+     */
+    ratio: {
+      type: [String, Number],
+      default: ART_PLAYER?.ratio || optionsDefault.ratio || undefined,
+      required: false,
+    },
+  },
+
+  setup(props) {
+    const option = {
+      ...deepMerge(ART_PLAYER, props),
+    };
+
+    option.src ??= {};
+
+    const src = {
+      ...deepMerge(optionsDefault.src, option.src),
+    };
+
+    src.url = option.url ?? src.url ?? "";
+    src.poster = option.poster ?? src.poster ?? "";
+    src.title = option.title ?? src.title ?? "";
+    src.muted = option.muted ?? src.muted ?? false;
+
+    const { el, width, height } = useSize<HTMLDivElement>(props, 0);
+
+    let artplayer: ArtPlayer;
+
+    const initArtPlayer = async (): Promise<void> => {
+      const { default: art } = await import("artplayer/dist/artplayer.js");
+
+      src.container = el.value!;
+      src.customType ??= {};
+      src.type ??= autoType(src.url!) || "";
+
+      if (src.type) {
+        switch (src.type) {
+          case "m3u8":
+          case "hls":
+            src.customType[src.type] = m3u8;
+            break;
+          case "flv":
+            src.customType[src.type] = flv;
+            break;
+          case "mpd":
+          case "dash":
+            src.customType[src.type] = mpd;
+            break;
+        }
+      }
+
+      if (props.pluginDanmuKu) {
+        const { default: artplayerPluginDanmuku } = await import(
+          "artplayer-plugin-danmuku"
+        );
+
+        src.plugins ??= [];
+        // @ts-ignore
+        src.plugins?.push(artplayerPluginDanmuku(option.pluginDanmuKu));
+      }
+
+      artplayer = new art(src as ArtPlayerOptions);
+
+      if (props.player) {
+        props.player(artplayer);
+      }
+    };
+
+    const destroyArtPlayer = (): void => {
+      artplayer.destroy();
+    };
```

Comment:
I do not think function body simple and meaningful like this, and only called once, should be extracted.

Reply:
> I do not think function body simple and meaningful like this, and only called once, should be extracted.

I want to add custom initialization and destroy, but I don't think it is necessary after thinking about it